caikit/core/modules/base.py

```diff
@@ -79,6 +79,33 @@ def metadata(self) -> Dict[str, Any]:
             self._metadata = {}
         return self._metadata
 
+    @property
```

Comment:
```suggestion
    @classmethod
    @property
```

Reply:
@gabe-l-hart see my comment in the docstring. Combing @property and @classmethod seemed to break the py3.8 tests. Using them together was also [deprecated in 3.11](https://docs.python.org/3.11/library/functions.html#classmethod)

Reply:
Read better Gabe! I know we have done this in places, so let me see what the right fully compatible way to do it is

Reply:
Ok, I think with this NOTE in the docstring, we're good. Thanks!